packages/esm-patient-forms-app/src/forms/form-view-see-all.component.tsx

```diff
@@ -0,0 +1,185 @@
+  if (currentVisit) {
+    const htmlForm = isHTMLForm(formUuid);
+    isEmpty(htmlForm)
+      ? launchWorkSpace(formUuid, patient, currentVisit?.uuid)
+      : navigate({
+          to: `\${openmrsBase}/htmlformentryui/htmlform/${htmlForm.UIPage}.page?patientId=${patient.id}&definitionUiResource=referenceapplication:htmlforms/${htmlForm.formAppUrl}.xml`,
+        });
+  } else {
+    startVisitPrompt();
+  }
+}
+
+const launchWorkSpace = (formUuid: string, patient: fhir.Patient, visitUuid?: string) => {
+  formEntrySub.next({ formUuid, patient, visitUuid });
+  attach('patient-chart-workspace-slot', 'patient-form-entry-workspace');
+};
+
+function isHTMLForm(formUuid: string) {
+  const htmlForms = CoreHTMLForms;
+  return htmlForms.find((form) => form.formUuid === formUuid);
+}
+
+interface FormViewProps {
+  forms: Array<Form>;
+  patientUuid: string;
+  patient: fhir.Patient;
+  encounterUuid?: string;
+}
+
+const filterFormsByName = (formName: string, forms: Array<Form>) => {
+  return forms.filter((form) => form.name.toLowerCase().search(formName.toLowerCase()) !== -1);
+};
+
+const FormView: React.FC<FormViewProps> = ({ forms, patientUuid, patient }) => {
+  const { t } = useTranslation();
+  const { currentVisit } = useVisit(patientUuid);
+  const [searchTerm, setSearchTerm] = React.useState<string>(null);
+  const [allForms, setAllForms] = React.useState<Array<Form>>(forms);
+  const { results, goTo, currentPage } = usePagination(allForms.sort(sortFormLatestFirst), 10);
+
+  const handleSearch = React.useMemo(() => debounce((searchTerm) => setSearchTerm(searchTerm), 300), []);
+
+  React.useEffect(() => {
+    setAllForms(!isEmpty(searchTerm) ? filterFormsByName(searchTerm, forms) : forms);
+  }, [searchTerm, forms]);
+
+  const tableHeaders: Array<DataTableHeader> = useMemo(
+    () => [
+      {
+        key: 'lastCompleted',
+        header: t('lastCompleted', 'Last Completed'),
+      },
+      { key: 'formName', header: t('formName', 'Form Name (A-Z)') },
+    ],
+    [t],
+  );
+
+  const tableRows: Array<DataTableRow> = useMemo(
+    () =>
+      results.map((form) => {
+        return {
+          id: form.uuid,
+          lastCompleted: form.lastCompleted && formatDate(form.lastCompleted),
+          formName: form.name,
+          formUuid: form.uuid,
+        };
+      }),
+    [results],
+  );
+
+  const withValue = (cell, row) => {
+    return (
+      <TableCell
+        style={{
+          color: first<DataTableCell>(row.cells).value ? `#0f62fe` : `#525252`,
+        }}
+        key={cell.id}>
+        {cell.value ? cell.value : `${t('never', 'Never')}`}
+      </TableCell>
+    );
+  };
+  return (
+    <div className={styles.formContainer}>
+      <Search
+        id="searchInput"
+        labelText=""
+        className={styles.formSearchInput}
+        placeholder={t('searchForForm', 'Search for a form')}
+        onChange={(evnt) => handleSearch(evnt.target.value)}
```

Comment:
If you change the definition of taking `value -> foo(value)` to `evt -> foo(evt.target.value)` then there is no need for the second function; and you gain performance.

So please change the definition of `handleSearch` and pass it in here directly (i.e., `onChange={handleSearch}`)

Reply:
This is also done.